inspector/src/components/sourceEditor/shaderSourceEditor.ts

```diff
@@ -0,0 +1,139 @@
+     */
+    public static Show(options: IShaderSourceEditorOptions) {
+        if (this._CurrentState) {
+            const popupWindow = (Popup as any)["shader-source-editor"];
+            if (popupWindow) {
+                popupWindow.close();
+            }
+        }
+
+        let hostElement = Popup.CreatePopup(
+            "Babylon.js Shader Source Editor",
+            "shader-source-editor",
+            1000,
+            800
+        )!;
+
+        const globalState = new SSEState();
+        globalState.shaderMaterial = options.shaderMaterial;
+        globalState.hostElement = hostElement;
+        globalState.hostDocument = hostElement.ownerDocument!;
+        globalState.hostWindow =  hostElement.ownerDocument!.defaultView!;
+
+        this.render(hostElement, options);
+
+        // Close the popup window when the page is refreshed or scene is disposed
+        const popupWindow = (Popup as any)["shader-source-editor"];
+        if (globalState.shaderMaterial && popupWindow) {
+            globalState.shaderMaterial.getScene().onDisposeObservable.addOnce(() => {
+                if (popupWindow) {
+                    popupWindow.close();
+                }
+            });
+            window.onbeforeunload = () => {
+                var popupWindow = (Popup as any)["shader-source-editor"];
+                if (popupWindow) {
+                    popupWindow.close();
+                }
+            };
+        }
+    }
+
+    static keyForMaterial(mat: ShaderMaterial): string[] {
+        const keys = [] as string[];
+        const shaderPath = mat.shaderPath;
+        if (typeof shaderPath === 'string') {
+            keys.push(shaderPath);
+        } else if (typeof shaderPath === 'object') {
+            if (typeof shaderPath.vertex === 'string') {
+                keys.push(shaderPath.vertex);
+            }
+            if (typeof shaderPath.fragment === 'string') {
+                keys.push(shaderPath.fragment);
+            }
+            if (typeof shaderPath.vertexElement === 'string') {
+                keys.push(shaderPath.vertexElement);
+            }
+            if (typeof shaderPath.fragmentElement === 'string') {
+                keys.push(shaderPath.fragmentElement);
+            }
+        }
+        return [ mat.name, ...keys ];
+    }
+
+    static render(hostElement: HTMLElement, options: IShaderSourceEditorOptions) {
+        const material = options.shaderMaterial;
+        const effect = material.getEffect();
+        const fragSource = (effect as any)?.originalFragmentSourceCode || '';
+        const vertexSource = (effect as any)?.originalVertexSourceCode || '';
+        const dataSource = new HistoryDataSource(this.keyForMaterial(material));
+        const delegate = {
+            recompile(fragSource: string, vertexSource: string, callback: ShaderSourceRecompileCallback) {
+                effect?._rebuildProgram(vertexSource, fragSource, (piplineContext) => {
```

Comment:
We cannot use _ function

@sebavan : Can you check if there is a way to rebuild or maybe just expose that feature?